nautobot/docs/plugins/development.md

```diff
@@ -114,37 +146,43 @@ Nautobot looks for the `config` variable within a plugin's `__init__.py` to load
+| `jobs`                | The dotted path to the list of Job classes (default: `jobs.jobs`)                                                                                                                       |
+| `max_version`         | Maximum version of Nautobot with which the plugin is compatible                                                                                                                         |
+| `menu_items`          | The dotted path to the list of menu items provided by the plugin (default: `navigation.menu_items`)                                                                                     |
+| `middleware`          | A list of middleware classes to append after Nautobot's built-in middleware                                                                                                             |
+| `min_version`         | Minimum version of Nautobot with which the plugin is compatible                                                                                                                         |
+| `name`                | Raw plugin name; same as the plugin's source directory                                                                                                                                  |
+| `required_settings`   | A list of any configuration parameters that **must** be defined by the user                                                                                                             |
+| `template_extensions` | The dotted path to the list of template extension classes (default: `template_content.template_extensions`)                                                                             |
+| `verbose_name`        | Human-friendly name for the plugin                                                                                                                                                      |
+| `version`             | Current release ([semantic versioning](https://semver.org/) is encouraged)                                                                                                              |
 
 All required settings must be configured by the user. If a configuration parameter is listed in both `required_settings` and `default_settings`, the default setting will be ignored.
 
 ### Install the Plugin for Development
 
-To ease development, it is recommended to go ahead and install the plugin at this point using setuptools' `develop` mode. This will create symbolic links within your Python environment to the plugin development directory. Call `setup.py` from the plugin's root directory with the `develop` argument (instead of `install`):
+To ease development, it is recommended with the virtual environment activated to use `poetry install` command from the location of the `pyproject.toml` file. By default Poetry installs the current project as editable and the dependencies as well.
```

Comment:
This could do with some clarification - what virtual environment is being referred to here?